### PR TITLE
Improve E2 interface field descriptions

### LINKED ISSUE #38: E2 fields -page improvements
On this page: https://paytraildocs.netlify.app/payments/e2-interface/fields/

- `AMOUNT`: payment maximum amount is **499 999,99 €** but there may be payment method specific limitations smaller than this.
- `AUTHCODE` could contain a link to the calculation page: https://paytraildocs.netlify.app/payments/e2-interface/authcode/
- `REFERENCE_NUMBER` should mention that single settlements are always made in the Finnish national reference standard, we do not make settlements using the RF reference format.
- `VAT_IS_INCLUDED` could contain a bit more of an explanation:

**Now:**

> Whether VAT is included in prices given in `ITEM` records. Available values **1** (VAT is included, default), **0** (VAT is not included). This field is only used with ITEM records. If brought without, an error message is shown.
>
> If using the Collector payment method, the value must be 1. Collector will not be available if the value is **0**.

**New:**

> Whether VAT is included in prices given in `ITEM` records. Available values **1** (VAT is included, default), **0** (VAT is not included). This field is only used with ITEM records. If brought without, an error message is shown.
>
> If `VAT_IS_INCLUDED = 0`, payment service will calculate the sum to be charged from the customer based on the `ITEM_UNIT_PRICE`, `ITEM_VAT_PERCENT`, `ITEM_QUANTITY`, and `ITEM_DISCOUNT_PERCENT` values. This may lead to rounding differences between the web shop and the payment service. Please note the calculation is done using two decimals.
>
> If using the Collector payment method, the value must be **1**. Collector will not be available if the value is **0**.

- Product Information contains info on the payment minimum sum, but it should also contain the maximum:

> Total sum of the product prices must be between **0.65 €** and **499 999.99 €**. There may be payment method specific limitations smaller than the maximum amount.

- Add info on which `ITEM_` fields need to be included in the data if any (not all of them are required I think).
- Add info that there can be max 500 products rows.

### PR DESCRIPTION
**Pull request by nikoheikkila**
## What type of Pull Request is this?

- [ ] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [x] Documentation Update

## Description

E2 interface field descriptions have been made more clear according to Customer Service wishes.

## Tests

- [ ] Not needed
- [ ] Unit tests added
- [ ] Integration tests added
- [x] Visual verification done

## Related Tickets & Documents

Fixes #38

## Code of Conduct

- [x] I have read and agreed to the [community code of conduct][coc]. The sole intention of this pull request is to improve the project codebase.

[coc]: https://github.com/paytrail/.github/blob/master/CODE_OF_CONDUCT.md

### CODE REVIEW

File: content/payments/e2-interface/fields.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ URL where customer is redirected after failed or cancelled payment. `PARAMS_OUT`
 Used to identify one transaction.
 
 #### `AMOUNT (*)`
-Price is given in euros and cents without currency. Decimals are separated with a dot. Must contain **two decimals**. Minimum price accepted by the service is **0.65 €**.
+Price is given in euros and cents without currency. Decimals are separated with a dot. Must contain **two decimals**. Interface accepts values between **0.65–499 999.99 €**. There may be payment method specific limitations smaller than the maximum amount.
```
Comment (Jipsukka) [inline content/payments/e2-interface/fields.md:37]: Probably would be good idea also mention that some payment methods might have larger minimum as well.

File: content/payments/e2-interface/fields.md [hunk 2]
```diff
@@ -118,27 +129,33 @@ Payer's company.
...
-#### `ITEM_TITLE[N]`
-Free field for product name. Required if product rows are included in data.
+Total sum of the product prices must be between **0.65 – 499 999.99 €**. There may be payment method specific limitations smaller than the maximum amount.
+
+#### `ITEM_TITLE[N] (*)`
+Free field for the product name.
 
 #### `ITEM_ID[N]`
-Product ID.
+Optional product ID.
```
Comment (Jipsukka) [inline content/payments/e2-interface/fields.md:143]: Maybe mention limitations, that it needs to be numeric only.

File: content/payments/e2-interface/fields.md (hunk 1 above)
Comment (nikoheikkila) [reply content/payments/e2-interface/fields.md]: True. Linked to the limitations page which should contain necessary info.

File: content/payments/e2-interface/fields.md (hunk 2 above)
Comment (nikoheikkila) [reply content/payments/e2-interface/fields.md:143]: Fix'd.

**Pull request merged**